src/lib/krb5/asn.1/asn1_encode.c

```diff
@@ -176,6 +176,8 @@ k5_asn1_decode_bool(const uint8_t *asn1, size_t len, intmax_t *val)
 {
     if (len != 1)
         return ASN1_BAD_LENGTH;
+    if (*asn1 && *asn1 != 0xFF)
+        return ASN1_BAD_FORMAT;
```

Comment:
That’s understandable.  At some point, would it be reasonable to add them if we can’t find any code that violates them?

Reply:
Could we make the behavior configurable?  Clients should be fine with strict checking, and it would be nice if servers logged warnings when they received invalid messages.

Reply:
I would prefer not to.  It would involve a lot of churn to pass context parameters into all the ASN.1 decoding functions, and it would add to a large soup of rarely-used config options.

(A trivial aside related to this code snippet but not to this particular question: the ASN.1 boolean type is almost completely unused by Kerberos specifications.  Only the Microsoft KERB-PA-PAC-REQUEST padata type uses it.)

Reply:
Could we at least have a way to get warnings when a bad message is received?  I would like to switch to strict checking at some point.

Reply:
Surfacing warnings from the ASN.1 decoder to the KDC or kadmind would be tough, and I wouldn't welcome the complexity.  However, this could be done purely outside the decoder: whenever a request is received, re-encode the whole thing and issue a warning if the encoding is different.  (There's a chance that this might be noisy because our structures don't always map neatly onto the ASN.1 schema, but I don't know if there would be a specific problem.)  I don't really welcome that complexity either, but it would wind up within dispatch functions which currently aren't super complicated, so maybe it wouldn't be so bad.

Reply:
I'm also against making this configurable.  In addition to the code complexity, it adds another thing support has to check for - "does it work with the laxer_asn1 setting", etc..

Strict compliance would be very nice in an ideal world, but unfortunately a lot of the software we're dealing with can't or won't be patched (e.g., proprietary upstream, or baked onto appliances).  I would rather not be in the position of being significantly more strict than Microsoft, for instance, so the testing of what their implementation does/doesn't accept is appreciated.